### ISSUE TITLE
Python to xml: Whitespace in method doc not preserved (MARKDOWN_SUPPORT=NO)

### ISSUE THREAD

**Issue by Stewori** [doxygen/doxygen#7884]
**Describe the bug**
Processing a Python code snippet like
```
class A(object):
    def __init__(self):
        """blah
        blah
                xyz
            blah
        """
        ...
```
results in an xml snippet
```
...
        <detaileddescription>
<para><verbatim>blah
blah
xyz
    blah
</verbatim> </para>
        </detaileddescription>
...
```
Note how the whitespace before "xyz" compared to "blah" is incorrect. This only occurs if there are equal or more spaces difference between them than one Python code indent. This issue seems to occur only in methods in classes, not in functions on module level.

**Expected behavior**
I expect the xml snippet to be
```
...
        <detaileddescription>
<para><verbatim>blah
blah
        xyz
    blah
</verbatim> </para>
        </detaileddescription>
...
```

**To Reproduce**
Find a Python file and doxygen config attached: [bugreport.zip](https://github.com/doxygen/doxygen/files/4850252/bugreport.zip)
The issue is demonstrated like mentioned above and with a more sophisticated example testing various whitespace amounts.
After running doxygen on the example, `xml/classexample_1_1A.xml` contains descriptions with messed up whitespace. Results in `xml/namespaceexample.xml` are fine (surprisingly), so it just happens inside classes.

**Version**
Tested with doxygen 1.8.11 from distribution and with self-compiled snapshot from 2020-06-29, 1.8.19 (d5916c06fba9667911a19cf658e7c5ab54daf72c).
Both produce the described problem.

`lsb_release -a`:
```
LSB Version:	core-9.20160110ubuntu0.2-amd64:core-9.20160110ubuntu0.2-noarch:security-9.20160110ubuntu0.2-amd64:security-9.20160110ubuntu0.2-noarch
Distributor ID:	LinuxMint
Description:	Linux Mint 18 Sarah
Release:	18
Codename:	sarah
```

**Additional context**
This issue hinders use of custom doc postprocessing on top of doxygen xml output. E.g. applying rst parsing is faulty if code samples with indentation are present in the doc.

I looked for duplicates but the bugtracker is difficult to search and has almost 2000 issues. Sorry if I overlooked a duplicate, pointers welcome! Maybe it is related to #7069 or #7072. Note that I tried to set various values for MARKDOWN_SUPPORT and TAB_SIZE and these do not seem to have any effect on the issue.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7893

**Comment by Stewori** [comment]
Thanks, for fixing this so quickly. I confirm that it is fixed in your branch. Even with deeper nesting, i.e. an inner class, it works well:
```
class A(object):
    def __init__(self):
        """blah
            blah
                xyz
            blah
        """
        pass

    class B(object):
        def methB(self):
            """blah
            xyz
             xyz
              xyz
               xyz
                xyz
                 xyz
                  xyz
                   xyz
                    xyz
                     xyz
                      xyz
                       xyz
                        xyz
                         xyz
            blah
            """
            pass
```

However, I did not check for regressions potentially caused by this fix; as far as I can tell it looks good to me.

**Comment by albert-github** [comment]
Code, with some updates, has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
